src/include/OpenImageIO/imageio.h

```diff
@@ -1025,7 +1025,10 @@ class OIIO_API ImageInput {
     ///         `true` upon success, or `false` upon failure.
     virtual bool valid_file (const std::string& filename) const;
 
-    /// Check valid vile using a UTF-16 encoded wstring filename.
+    /// Check valid file using an IOProxy.
+    virtual bool valid_file (Filesystem::IOProxy* io) const;
+
+    /// Check valid file using a UTF-16 encoded wstring filename.
```

Comment:
Would you mind, here, (a) moving the new one after the next method (so the two that take filenames are next to each other), and (b) replicating the full documentation comment for the IOProxy one, of course adjusted to describe the new argument?

Reply:
Sure thing! I also noticed that the public API here preferred to use `ioproxy` as the parameter name so I changed everything from `io` to `ioproxy` for consistency.

Reply:
perfect, thanks